### PR TITLE
Add financial example with xgboost

### PR DESCRIPTION
**Pull request by ZiyueXu77**
Add financial covering all xgboost methods: vertical, horizontal_histogram/cyclic/tree, also update xgboost tree server code to account for xgboost's recent API updates 

Fixes # .

### Description

A few sentences describing the changes proposed in this pull request.

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

File: examples/advanced/finance/README.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+This will cover baseline centralized training, horizontal FL with histogram-based, tree-based cyclic, and tree-based bagging
+collaborations, as well as vertical FL.
+
+Then we test the resulting models on the test dataset with 
+```
+bash run_testing.sh
+``` 
+The results are as follows:
+```
+Testing baseline_xgboost.py
```
Comment (holgerroth) [inline examples/advanced/finance/README.md:57]: Why is this line showing *.py?

File: examples/advanced/finance/run_testing.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+echo "Testing baseline_xgboost.py"
```
Comment (holgerroth) [inline examples/advanced/finance/run_testing.sh:2]: remove .py

**Review by holgerroth — COMMENTED** [review]
Rather than adding all these new config files, should we be adding xgboost to the [job templates](https://github.com/NVIDIA/NVFlare/tree/main/job_templates) and then use the CLI to change specific values?

File: examples/advanced/finance/run_testing.sh (hunk 2 above)
Comment (ZiyueXu77) [reply examples/advanced/finance/run_testing.sh]: removed

**Comment by ZiyueXu77** [comment]
> Rather than adding all these new config files, should we be adding xgboost to the [job templates](https://github.com/NVIDIA/NVFlare/tree/main/job_templates) and then use the CLI to change specific values? Seems this [PR](https://github.com/NVIDIA/NVFlare/pull/2066) already adds xgboost job templates which could be used here.

Sounds good! Let's wait for this PR to merge and refactor

**Comment by SYangster** [comment]
> Sounds good! Let's wait for this PR to merge and refactor

Yes I agree the templates would be a good way to reuse those configurations. Right now I moved the vertical xgboost examples to the templates, however I haven't converted the regular horizontal ones yet. Those would require using `prepare_job_config.py` to generate the various configurations, do we want to do that?

File: examples/advanced/finance/README.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+AUC score:  0.965017768854869
+Testing xgboost_vertical
+AUC score:  0.8822601617321099
```
Comment (SYangster) [inline examples/advanced/finance/README.md:60]: Maybe here we add both the 0.5 and the 1.0 overlap versions to compare performance for vertical?

Comment (ZiyueXu77) [reply examples/advanced/finance/README.md]: just updated readme to 1.0 overlap, making the settings identical among all experiments

**Comment by ZiyueXu77** [comment]
> > Sounds good! Let's wait for this PR to merge and refactor
> 
> Yes I agree the templates would be a good way to reuse those configurations. Right now I moved the vertical xgboost examples to the templates, however I haven't converted the regular horizontal ones yet. Those would require using `prepare_job_config.py` to generate the various configurations, do we want to do that?

right, the settings can have some variations @holgerroth what do you think?

**Review by YuanTingHsieh — APPROVED** [review]
We can merge this for now, enhancement PR can come later

File: examples/advanced/finance/requirements.txt [hunk 4]
```diff
@@ -0,0 +1,7 @@
+nvflare>=2.3.0
+openmined.psi==1.1.1
+pandas
+xgboost>=1.7.0
```
Comment (YuanTingHsieh) [inline examples/advanced/finance/requirements.txt:4]: let's update this version later with xgboost example.

Comment (ZiyueXu77) [reply examples/advanced/finance/requirements.txt:4]: sure

**Comment by holgerroth** [comment]
> We can merge this for now, enhancement PR can come later

I create a new issue for the enhancement https://github.com/NVIDIA/NVFlare/issues/2079

**Pull request merged**